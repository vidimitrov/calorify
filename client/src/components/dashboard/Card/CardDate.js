import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import moment from 'moment';

const Wrapper = styled.div`
  color: #f0cd61;
  margin: 0px 20px;
  max-width: 100px;
`;

const FormattedDate = styled.h3`
  margin: 0;
  color: #c7b26e;

  @media (max-width: 420px) {
    font-size: 18px;
  }
`;

const CardDate = ({ date }) => {
  const formattedDate = moment(date).calendar();

  return (
    <Wrapper>
      <FormattedDate>{formattedDate}</FormattedDate>
    </Wrapper>
  );
};

CardDate.propTypes = {
  date: PropTypes.string.isRequired,
};

export default CardDate;
