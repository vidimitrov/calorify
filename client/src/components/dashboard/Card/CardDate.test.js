import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import CardDate from './CardDate';

it('Dashboard/Card/CardDate component renders correctly', () => {
  const tree = renderer.create(<CardDate date="2018-11-10T13:22" />).toJSON();
  expect(tree).toMatchSnapshot();
});
