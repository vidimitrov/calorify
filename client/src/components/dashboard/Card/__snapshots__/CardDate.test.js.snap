// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard/Card/CardDate component renders correctly 1`] = `
.c0 {
  color: #f0cd61;
  margin: 0px 20px;
  max-width: 100px;
}

.c1 {
  margin: 0;
  color: #c7b26e;
}

@media (min-width:320px) and (max-width:360px) {
  .c0 {
    min-width: 60px;
    margin: 0px 5px;
  }
}

@media (min-width:360px) and (max-width:420px) {
  .c0 {
    min-width: 70px;
    margin: 0px 5px;
  }
}

@media (max-width:420px) {
  .c1 {
    font-size: 16px;
  }
}

<div
  className="c0"
>
  <h3
    className="c1"
  >
    Today at 1:22 PM
  </h3>
</div>
`;
